File: Yang params model/Yang params model/psi1_b.m
%Marini 
function psi1B = psi1_b(day)
    global P 
    %Coordinates used for coefficient fit
    %x_values = [10, 15, 25, 30];
    %y_values = [2, 4.5, 7.4, 6.7];
    coefficients = [-0.0007, 0.0227, 0.2817, -2.3500];
    x_values=temp(day, 0);
    psi1B = polyval(coefficients, x_values);
    %plot(linspace(13, 33, 40), polyval((coefficients), (linspace(13, 33, 40))))
end

File: Yang params model/dengue_model/psi1_b.py
"""Tasa de desarrollo huevo → larva de *A. albopictus* (psi1_b).

Especie: A. albopictus.
Etapas: huevo → L1 (primera larva).
Unidad: día⁻¹.
Ajuste: polinomio grado 3 en T (°C).
Fuente: Marini et al.
Rango de ajuste: T ∈ [15, 35] °C.
"""
from __future__ import annotations

import numpy as np

from ._utils import clip_temp_for_poly, return_like_input
from .temp import temp


COEFFICIENTS = np.array([-0.0007, 0.0227, 0.2817, -2.3500], dtype=float)


def psi1_b(day):
    x_values = temp(day, 0)
    x_clipped = clip_temp_for_poly(x_values, source="psi1_b")
    result = np.polyval(COEFFICIENTS, x_clipped)
    return return_like_input(day, result)
